lib/appsignal/extension/jruby.rb

```diff
@@ -242,14 +242,15 @@ def self.lib_extension
           [:pointer],
           :appsignal_string
 
-        Appsignal.extension_loaded = true
+        Appsignal.extension_loaded = true if Appsignal.respond_to? :extension_loaded=
       rescue LoadError => error
         error_message = "ERROR: AppSignal failed to load extension. " \
           "Please run `appsignal diagnose` and email us at support@appsignal.com\n" \
           "#{error.class}: #{error.message}"
-        Appsignal.logger.error(error_message)
+        Appsignal.logger.error(error_message) if Appsignal.respond_to? :logger
```

Comment:
When could `Appsignal.respond_to? :logger` fail?

Reply:
During the extension installation, we don't load the entire Ruby gem, we don't load `lib/appsignal.rb`. When `ext/Rakefile` loads `lib/appsignal/extension/jruby.rb` `Appsignal.logger` would not be present and result in an error prevented by this conditional.

Reply:
Thank you for explaining that! 🙇